reconcile/utils/terrascript_aws_client.py

```diff
@@ -781,7 +781,9 @@ def populate_route53(self, desired_state, default_ttl=300):
             acct_name = zone["account_name"]
 
             # Ensure zone is in the state for the given account
-            zone_id = safe_resource_id(f"{zone['name']}")
+            # zone_id refers to the terraform identifier, not the AWS
+            # route 53 zone ID that is generated after the zone is created
+            zone_id = safe_resource_id(f"{zone['id']}")
```

Comment:
```suggestion
            zone_res_name = safe_resource_id(f"{zone['resource_name']}")
```

And then you could probably remove all 3 comments above here (i think the original one is something leftover?)